apps/utils/src/deploy.rs

```diff
@@ -1,14 +1,23 @@
-use essential_node_types::register_contract_solution;
-use essential_types::{contract::Contract, solution::Solution, ContentAddress};
+use essential_node_types::{register_contract_solution, register_program_solution};
+use essential_types::{
+    contract::Contract,
+    solution::{Solution, SolutionSet},
+    ContentAddress, Program,
+};
 
 pub async fn deploy_contract(
```

Comment:
I suggest a function that registers both a contract and its programs for app utils. In the node, we are taking a different approach of registering contracts and programs separately, but I think for apps it will be most useful to have a function like:

```suggestion    
pub async fn register_contract_and_programs(
    builder_conn: &ConnectionPool,
    contract_registry: &PredicateAddress,
    program_registry: &PredicateAddress,
    contract: &Contract,
    programs: Vec<Program>,
) -> anyhow::Result<ContentAddress> {
    let solutions = register_contract_solution(contract_registry.clone(), contract)
        .into_iter()
        .chain(
            programs
                .iter()
                .map(|p| register_program_solution(program_registry.clone(), p)),
        )
        .collect();
    let solution_set = Arc::new(SolutionSet { solutions });
    let r = builder_conn
        .insert_solution_set_submission(
            solution_set,
            SystemTime::now().duration_since(UNIX_EPOCH).unwrap(),
        )
        .await?;
    Ok(r)
}
```

Even if that is not the way to go, at the very least I suggest not constructing the default big bang multiple times. This can be achieved by passing necessary big bang fields to relevant functions.

Reply:
I feel like devs shouldn't need to even think about deploying programs... it's almost an implementation detail (by this I mean that in Pint, there's no `program` keyword; we just write contracts, with predicates, constraints and state reads).
I do think one function should achieve both deployment of the contract and its associated programs, just not sure if devs should even have to know about the program part.

Thoughts?

Reply:
I think it is not unreasonable to expect devs to know that:
- Solution sets contain solutions that point to predicates via their addresses in predicate registry contract's state. Those addresses should be __valid__ predicate addresses.
- Similarly, predicates point to programs via their addresses in program registry contract's state. Those addresses should be __valid__ program addresses.

Note: __Valid__ here means that the corresponding predicate / program is either already registered, or is being registered in the solution that attempts to use it.

But I see your point that pint abstracts programs away. The protocol is built on them though 🤔